File: BinarySearchTree/Java/src/BinarySearchTree.java
public class BinarySearchTree {
    Node root;

    public void insert(Node node) {
        root = insertHelper(root, node);
    }
    private Node insertHelper(Node root, Node node) {
        int data = node.data;

        if (root == null) {
            root = node;
            return root;
        }
        else if (data < root.data) {
            root.left = insertHelper(root.left, node);
        }
        else {
            root.right = insertHelper(root.right, node);
        }
        return root;
    }

    public void display() {
        displayHelper(root);
    }
    private void displayHelper(Node root) {
        if (root == null) {
            return;
        }
        displayHelper(root.left);
        System.out.println(root.data);
        displayHelper(root.right);
    }
}

File: BinarySearchTree/GoLang/BinarySearchTree.go
package main

import "fmt"

type BinarySearchTree struct {
	root *Node
}

func (tree *BinarySearchTree) Insert(data int) {
	tree.root = tree.insertHelper(tree.root, data)
}

func (tree *BinarySearchTree) insertHelper(root *Node, data int) *Node {
	if root == nil {
		root = &Node{data: data}
		return root
	} else if root.data > data {
		root.left = tree.insertHelper(root.left, data)
	} else {
		root.right = tree.insertHelper(root.right, data)
	}
	return root
}

func (tree *BinarySearchTree) Display() {
	tree.displayHelper(tree.root)
}

func (tree *BinarySearchTree) displayHelper(root *Node) {
	if root == nil {
		return
	}
	tree.displayHelper(root.left)
	fmt.Println(root.data)
	tree.displayHelper(root.right)
}
